management/persistence.md

```diff
@@ -0,0 +1,330 @@
+
+## RDB advantages
+
+* RDB is a very compact single-file point-in-time representation of your FalkorDB data. RDB files are perfect for backups. For instance you may want to archive your RDB files every hour for the latest 24 hours, and to save an RDB snapshot every day for 30 days. This allows you to easily restore different versions of the data set in case of disasters.
+* RDB is very good for disaster recovery, being a single compact file that can be transferred to far data centers, or onto Amazon S3 (possibly encrypted).
+* RDB maximizes FalkorDB performances since the only work the FalkorDB parent process needs to do in order to persist is forking a child that will do all the rest. The parent process will never perform disk I/O or alike.
+* RDB allows faster restarts with big datasets compared to AOF.
+* On replicas, RDB supports partial resynchronizations after restarts and failovers.
+
+## RDB disadvantages
+
+* RDB is NOT good if you need to minimize the chance of data loss in case FalkorDB stops working (for example after a power outage). You can configure different *save points* where an RDB is produced (for instance after at least five minutes and 100 writes against the data set, you can have multiple save points). However you'll usually create an RDB snapshot every five minutes or more, so in case of FalkorDB stopping working without a correct shutdown for any reason you should be prepared to lose the latest minutes of data.
+* RDB needs to fork() often in order to persist on disk using a child process. fork() can be time consuming if the dataset is big, and may result in FalkorDB stopping serving clients for some milliseconds or even for one second if the dataset is very big and the CPU performance is not great. AOF also needs to fork() but less frequently and you can tune how often you want to rewrite your logs without any trade-off on durability.
+
+## AOF advantages
+
+* Using AOF FalkorDB is much more durable: you can have different fsync policies: no fsync at all, fsync every second, fsync at every query. With the default policy of fsync every second, write performance is still great. fsync is performed using a background thread and the main thread will try hard to perform writes when no fsync is in progress, so you can only lose one second worth of writes.
+* The AOF log is an append-only log, so there are no seeks, nor corruption problems if there is a power outage. Even if the log ends with a half-written command for some reason (disk full or other reasons) the FalkorDB-check-aof tool is able to fix it easily.
+* FalkorDB is able to automatically rewrite the AOF in background when it gets too big. The rewrite is completely safe as while FalkorDB continues appending to the old file, a completely new one is produced with the minimal set of operations needed to create the current data set, and once this second file is ready FalkorDB switches the two and starts appending to the new one.
```

Comment:
No need to explain how the mechanism work, just say that AOF don't grow infinitely, from time to time the AOF is compacted by including a snapshot of the dataset followed by the latest write commands.